subprojects/jvm-services/src/main/java/org/gradle/internal/jvm/inspection/DefaultJvmMetadataDetector.java

```diff
@@ -101,11 +99,12 @@ private JvmInstallationMetadata asMetadata(File javaHome, EnumMap<ProbedSystemPr
             }
             return error("Command returned unexpected result code: " + exitValue + "\nError output:\n" + errorOutput);
         } catch (ExecException ex) {
-            return error(ex.getMessage());
+            return error(Throwables.getStackTraceAsString(ex));
```

Comment:
:x: I don't think we should change error handling this way.

Instead of capturing the whole stacktrace as text here deep into the infratructure, I would capture the actual exception so it can be sanitized/analyzed/reported in the same way as other exceptions thrown by Gradle. Then other components that later decide to fail the build because of problems probing jvm installations can attach the actual exceptions, maybe as suppressed exceptions, to a contextual one. This would make error reporting uniform with what we do elsewhere while providing all the information required to troubleshoot probing issues.

I'd be fine if refining error handling was done in a separate PR.

Reply:
Pulling this out into a separate PR